### PR TITLE
Replaces orcid client with gem.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #3241

## Why was this change made?
Remove redundant code.

## How was this change tested?
Unit, QA

```
pub@sul-pub-cap-dev-a:~/sul-pub/current$ RAILS_ENV=production bin/rake orcid:add_all_works
Added 52 works from 12 ORCID users.
```

### CODE REVIEW

File: config/initializers/orcid_client.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+SulOrcidClient.configure(
+  client_id: Settings.ORCID.CLIENT_ID,
+  client_secret: Settings.ORCID.CLIENT_SECRET,
+  base_url: Settings.ORCID.BASE_URL,
+  base_public_url: Settings.ORCID.BASE_PUBLIC_URL,
+  base_auth_url: Settings.ORCID.BASE_AUTH_URL
```
Comment (mjgiarlo) [inline config/initializers/orcid_client.rb:6]: We typically use all lowercase to name Settings variables, but I'm assuming this PR is piggybacking off of existing shared_configs (that is still uppercased)? In which case I leave it to you whether it's timely or not to make this tweak.

File: lib/orcid.rb [hunk 2]
```diff
@@ -25,7 +25,7 @@ def self.base_orcidid(orcidid)
 
   # Fetch works for a known ID for which we expect to get back publications
   def self.working?
-    response = client.fetch_works(Settings.ORCID.orcidid_for_check)
+    response = client.fetch_works(orcidid: Settings.ORCID.orcidid_for_check)
```
Comment (mjgiarlo) [inline lib/orcid.rb:28]: Trivial: consider `orcid_id` ❓

File: config/initializers/orcid_client.rb (hunk 1 above)
Comment (justinlittman) [reply config/initializers/orcid_client.rb:6]: Yeah, this is following the existing conventions in sul_pub and using existing configuration.

File: lib/orcid.rb (hunk 2 above)
Comment (mjgiarlo) [reply lib/orcid.rb:28]: Totally fine not to tangle this change with a large refactoring/renaming 👍🏻

**Review by mjgiarlo — APPROVED** [review]
Thumbs-up from me, but would like @peetucket to take a quick gander as well.

**Review by peetucket — APPROVED** [review]

**Pull request merged**